app/containers/RoomHeader/RoomHeader.js

```diff
@@ -41,31 +41,70 @@ const styles = StyleSheet.create({
 	}
 });
 
-const SubTitle = React.memo(({
-	usersTyping, subtitle, renderFunc, theme, scale
-}) => {
-	const fontSize = getSubTitleSize(scale);
-	// typing
-	if (usersTyping.length) {
+const ActivityIndicator = React.memo(({ users, action }) => {
+	const getUsersText = (performingUsers = []) => {
 		let usersText;
-		if (usersTyping.length === 2) {
-			usersText = usersTyping.join(` ${ I18n.t('and') } `);
+		if (performingUsers.length === 2) {
+			usersText = performingUsers.join(` ${ I18n.t('and') } `);
 		} else {
-			usersText = usersTyping.join(', ');
+			usersText = performingUsers.join(', ');
 		}
-		return (
-			<Text style={[styles.subtitle, { fontSize, color: themes[theme].auxiliaryText }]} numberOfLines={1}>
-				<Text style={styles.typingUsers}>{usersText} </Text>
-				{ usersTyping.length > 1 ? I18n.t('are_typing') : I18n.t('is_typing') }...
-			</Text>
-		);
+		return usersText;
+	};
+
+	return (
+		<Text>
+			<Text style={[styles.subtitle, styles.typingUsers]}>{getUsersText(users)} </Text>
+			{ users.length > 1 ? I18n.t(`are_${ action }`) : I18n.t(`is_${ action }`) }...
+		</Text>
+	);
+});
+
+ActivityIndicator.propTypes = {
+	action: PropTypes.string,
+	users: PropTypes.array
+};
+
+const UserActivity = React.memo(({ activities, tmid, rid }) => {
+	const { typing, uploading, recording } = activities;
+	const id = !tmid ? rid : tmid;
```

Comment:
You can simplify to:
```suggestion
	const id = tmid || rid;
```